app/services/companies_list_updator.rb

```diff
@@ -0,0 +1,14 @@
+# frozen_string_literal: true
+
+class CompaniesListUpdator < ApplicationService
+  def call
+    companies = GPW::Client.new.fetch_companies_list
+
+    companies.each do |company|
+      Company.find_by(isin: company.fetch(:isin))&.update!(
+        name: company.fetch(:name),
+      )
+      Company.find_or_create_by!(company)
+    end
```

Comment:
You could extract part of the code to private method here, for example:

```suggestion
    companies_attrs.each { |company_attrs| upsert_company(company_attrs) }
```

Reply:
Good suggestion, I expanded my knowledge of object attributes :)